org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/config/generatePomXml.mtl

```diff
@@ -172,20 +176,22 @@
         [generalDependencies()/]
         [servletDependencies(true)/]
         [lyoDependencies(true, not anAdaptorInterface.requiredAdaptors->isEmpty(), anAdaptorInterface.authenticationEnabled())/]
+        [if (anAdaptorInterface.swaggerDocumentationEnabled())]
         <!-- Swagger dependencies -->
         <dependency>
             <groupId>io.swagger</groupId>
             <artifactId>swagger-jersey2-jaxrs</artifactId>
-            <version>1.5.22</version>
+            <version>${swagger.version}</version>
         </dependency>
+        [/if]
     </dependencies>
     <!-- [protected ('post_dependencies')]
     -->
     <!-- [/protected]
     -->
     <build>
-        <finalName>[mavenProjectConfiguration.artifactId/]</finalName>
-            <plugins>
+        <finalName>[mavenServerConfiguration.applicationContextPath/]</finalName>
```

Comment:
I have '/' as the context path in my adaptors (I deploy only one adaptor per servlet container), I believe this will not be good here.

Reply:
Good point. What should be done if no context path is given.
Skip the whole finalName entry? 

Have you otherwise managed to make an Adaptor work without a context path? I recall I had problems. Would be interesting to document how to.

Reply:
Skipping is a good idea.

Reply:
I noticed that original code did not even work in the most basic case, where one has "/adaptor-name". I need to remove the /.
Please check my second commit, which will solve your and my issues.